File: 프로그래머스/Level1/실패율.py
# https://programmers.co.kr/learn/courses/30/lessons/42889

def solution(N, stages):
    temp = []
    len_stages = len(stages)
    for i in range(1, N + 1):
        cnt = stages.count(i)
        if cnt == 0:
            fail = 0
        else:
            fail = cnt / len_stages
        temp.append((i, fail))
        len_stages -= cnt    
    temp.sort(key=lambda x:x[1], reverse=True)
    answer = []
    for a, b in temp:
        answer.append(a)
    return answer

File: 프로그래머스/Level1/실패율.js
// https://programmers.co.kr/learn/courses/30/lessons/42889

const solution = (N, stages) => {
	const answer = [];
	for (let i = 1; i <= N; i++) {
		const total = stages.filter((a) => a >= i).length;
		const fail = stages.filter((a) => a === i).length;
		answer.push([fail / total, i]);
	}
	return answer.sort((a, b) => b[0] - a[0]).map((a) => a[1]);
};
